src/genie/libs/parser/iosxr/show_interface.py

```diff
@@ -3032,4 +3032,72 @@ def cli(self, interface=None, output=None):
         else:
             out = output
 
-        return super().cli(output=out)
\ No newline at end of file
+        return super().cli(output=out)
+
+
+#############################################################################
+# Parser For show interface summary
+#############################################################################
+
+class ShowInterfaceSummarySchema(MetaParser):
+    """schema for show interface summary
+    """
+
+    schema = {
+        'interface_types': {
+            Any(): {
+                'total': str,
+                'up': str,
+                'down': str,
+                'admin_down': str,
+            }
+        }
+    }
+
+
+class ShowInterfaceSummary(ShowInterfaceSummarySchema):
+    """parser for show interface summary
+    """
+
+    cli_command = ['show interface summary']
+
+    def cli(self, interface="", output=None):
+        if output is None:
+            if interface:
+                cmd = self.cli_command[1].format(interface=interface)
+            else:
+                cmd = self.cli_command[0]
+            out = self.device.execute(cmd)
+        else:
+            out = output
+
+        interface_dict = {}
+        ret_dict = {}
+
+        # <INTERFACE_TYPE_NAME> 60 20 0 1
```

Comment:
Comment examples should always be exact lines form the output that can be placed directly into regex tools for testing.

```suggestion
        # IFT_FORTYGETHERNET      2        1        0        1
```